cmd/snap-bootstrap/partition/encrypt.go

```diff
@@ -133,3 +161,61 @@ func cryptsetupClose(name string) error {
 	}
 	return nil
 }
+
+func cryptsetupAddKey(key EncryptionKey, rkey RecoveryKey, node string) error {
+	// create a named pipe to pass the recovery key
```

Comment:
The basic principle of how keys are handled seems good.

Is this code something that should pass security review?

Reply:
It should, I'll ask @chrisccoulson to review it.

Reply:
This looks ok - I do think some of this code should probably live in snapcore/secboot along with the other cryptsetup helpers, so it would provide APIs for creating volumes, setting / updating keys.